test/e2e/node/pod_hostnamefqdn.go

```diff
@@ -0,0 +1,146 @@
+				{
+					Name:  "test-container",
+					Image: imageutils.GetE2EImage(imageutils.BusyBox),
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	return pod
+}
+
+var _ = SIGDescribe("Hostname of Pod [Feature:SetHostnameAsFQDN]", func() {
+	f := framework.NewDefaultFramework("hostfqdn")
+
+	/*
+	   Release : v1.19
+	   Testname: Create Pod without fully qualified domain name (FQDN)
+	   Description: A Pod that does not define the subdomain field in it spec, does not have FQDN.
+	*/
+	ginkgo.It("a pod without subdomain field does not have FQDN [Feature:SetHostnameAsFQDN]", func() {
+		pod := testPod()
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "echo $(hostname)';'$(hostname -f)"}
+		output := []string{fmt.Sprintf("%s;%s", pod.ObjectMeta.Name, pod.ObjectMeta.Name)}
+		// Create Pod
+		f.TestContainerOutput("shotname only", pod, 0, output)
+	})
+
+	/*
+	   Release : v1.19
+	   Testname: Create Pod without FQDN, setHostnameAsFQDN field set to true
+	   Description: A Pod that does not define the subdomain field in it spec, does not have FQDN.
+	                Hence, SetHostnameAsFQDN feature has no effect.
+	*/
+	ginkgo.It("a pod without FQDN is not affected by SetHostnameAsFQDN field [Feature:SetHostnameAsFQDN]", func() {
+		pod := testPod()
+		// Setting setHostnameAsFQDN field to true should have no effect.
+		setHostnameAsFQDN := true
+		pod.Spec.SetHostnameAsFQDN = &setHostnameAsFQDN
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "echo $(hostname)';'$(hostname -f)"}
+		output := []string{fmt.Sprintf("%s;%s", pod.ObjectMeta.Name, pod.ObjectMeta.Name)}
+		// Create Pod
+		f.TestContainerOutput("shotname only", pod, 0, output)
+	})
+
+	/*
+	   Release : v1.19
+	   Testname: Create Pod with FQDN, setHostnameAsFQDN field not defined.
+	   Description: A Pod that defines the subdomain field in it spec has FQDN.
+	                hostname command returns shortname (pod name in this case), and hostname -f returns FQDN.
+	*/
+	ginkgo.It("a pod with subdomain field has FQDN, hostname is shortname [Feature:SetHostnameAsFQDN]", func() {
+		pod := testPod()
+		pod.Spec.Containers[0].Command = []string{"sh", "-c", "echo $(hostname)';'$(hostname -f)"}
+		subdomain := "t"
+		// Set PodSpec subdomain field to generate FQDN for pod
+		pod.Spec.Subdomain = subdomain
+		// Expected Pod FQDN
+		hostFQDN := fmt.Sprintf("%s.%s.%s.svc.%s", pod.ObjectMeta.Name, subdomain, f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		output := []string{fmt.Sprintf("%s;%s", pod.ObjectMeta.Name, hostFQDN)}
+		// Create Pod
+		f.TestContainerOutput("shotname and fqdn", pod, 0, output)
+	})
+
+	/*
+		   Release : v1.19
+		   Testname: Create Pod with FQDN, setHostnameAsFQDN field set to true.
+		   Description: A Pod that defines the subdomain field in it spec has FQDN. When setHostnameAsFQDN: true, the
+				hostname is set to be the FQDN. In this case, both commands hostname and hostname -f return
+				the FQDN of the Pod.
```

Comment:
```suggestion
	   Release : v1.19
	   Testname: Create Pod with FQDN, setHostnameAsFQDN field set to true.
	   Description: A Pod that defines the subdomain field in it spec has FQDN. When setHostnameAsFQDN: true, the
				hostname is set to be the FQDN. In this case, both commands hostname and hostname -f return
				the FQDN of the Pod.
```

Reply:
yes, I knew about this one and i could not get gomft happy, i think i finally figured out.